// carbon-components-react/lib/components/ToggleSmall/ToggleSmall-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Untoggled = exports.Toggled = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _ToggleSmall = _interopRequireDefault(require("../ToggleSmall"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var toggleProps = function toggleProps() {
  return {
    labelText: (0, _addonKnobs.text)('Label toggle input control (labelText)', 'Toggle element label'),
    className: 'some-class',
    labelA: (0, _addonKnobs.text)('Label for untoggled state (labelA)', ''),
    labelB: (0, _addonKnobs.text)('Label for toggled state (labelB)', ''),
    disabled: (0, _addonKnobs.boolean)('Disabled (disabled)', false),
    onChange: (0, _addonActions.action)('onChange'),
    onToggle: (0, _addonActions.action)('onToggle')
  };
};

var _default = {
  title: 'ToggleSmall',
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    component: _ToggleSmall.default,
    subcomponents: {}
  }
};
exports.default = _default;

var Toggled = function Toggled() {
  return /*#__PURE__*/_react.default.createElement(_ToggleSmall.default, _extends({
    defaultToggled: true
  }, toggleProps(), {
    className: "some-class",
    id: "toggle-1"
  }));
};

exports.Toggled = Toggled;
Toggled.storyName = 'toggled';
Toggled.parameters = {
  info: {
    text: "\n        Toggles are controls that are used to quickly switch between two possible states. The example below shows\n        an uncontrolled Toggle component. To use the Toggle component as a controlled component, set the toggled property.\n        Setting the toggled property will allow you to change the value dynamically, whereas setting the defaultToggled\n        prop will only set the value initially. This example has defaultToggled set to true. Small toggles may be used\n        when there is not enough space for a regular sized toggle. This issue is most commonly found in tables.\n      "
  }
};

var Untoggled = function Untoggled() {
  return /*#__PURE__*/_react.default.createElement(_ToggleSmall.default, _extends({}, toggleProps(), {
    className: "some-class",
    id: "toggle-1"
  }));
};

exports.Untoggled = Untoggled;
Untoggled.storyName = 'untoggled';
Untoggled.parameters = {
  info: {
    text: "\n        Toggles are controls that are used to quickly switch between two possible states. The example below shows\n        an uncontrolled Toggle component. To use the Toggle component as a controlled component, set the toggled property.\n        Setting the toggled property will allow you to change the value dynamically, whereas setting the defaultToggled\n        prop will only set the value initially. Small toggles may be used when there is not enough space for a regular sized toggle. This issue is most\n        commonly found in tables.\n      "
  }
};
