// carbon-components-react/lib/components/FormLabel/FormLabel-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithTooltip = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _FormLabel = _interopRequireDefault(require("./FormLabel"));

var _Tooltip = _interopRequireDefault(require("../Tooltip"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
var additionalProps = {
  className: 'some-class'
};
var _default = {
  title: 'FormLabel',
  parameters: {
    component: _FormLabel.default
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_FormLabel.default, additionalProps, "Label");
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: 'Form label.'
  }
};

var WithTooltip = function WithTooltip() {
  return /*#__PURE__*/_react.default.createElement(_FormLabel.default, additionalProps, /*#__PURE__*/_react.default.createElement(_Tooltip.default, {
    triggerText: "Label"
  }, "This is the content of the tooltip."));
};

exports.WithTooltip = WithTooltip;
WithTooltip.storyName = 'With tooltip';
WithTooltip.parameters = {
  info: {
    text: 'Form label with tooltip.'
  }
};
