// carbon-components-react/lib/components/RadioButton/RadioButton-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Skeleton = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _RadioButton = _interopRequireDefault(require("../RadioButton"));

var _RadioButton2 = require("./RadioButton");

var _RadioButton3 = _interopRequireDefault(require("../RadioButton/RadioButton.Skeleton"));

var _RadioButton4 = _interopRequireDefault(require("./RadioButton.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var labelPositions = {
  'Left (left)': 'left',
  'Right (right)': 'right'
};

var radioProps = function radioProps() {
  return {
    className: 'some-class',
    name: (0, _addonKnobs.text)('Form item name (name)', 'test'),
    value: (0, _addonKnobs.text)('Value (value)', 'standard'),
    labelText: (0, _addonKnobs.text)('Label text (labelText)', 'Standard Radio Button'),
    labelPosition: (0, _addonKnobs.select)('Label position (labelPosition)', labelPositions, 'right'),
    disabled: (0, _addonKnobs.boolean)('Disabled (disabled)', false),
    onChange: (0, _addonActions.action)('onChange')
  };
};

var _default = {
  title: 'RadioButton',
  decorators: [_addonKnobs.withKnobs],
  component: _RadioButton2.RadioButton,
  subcomponents: {
    RadioButtonSkeleton: _RadioButton3.default
  },
  parameters: {
    docs: {
      page: _RadioButton4.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_RadioButton.default, _extends({
    id: "radio-1"
  }, radioProps()));
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n            Radio buttons are used when a list of two or more options are mutually exclusive,\n            meaning the user must select only one option. The example below shows how the Radio Button component\n            can be used as an uncontrolled component that is initially checked by setting the defaultChecked property\n            to true. To use the component in a controlled way, set the checked property instead.\n          "
  }
};

var Skeleton = function Skeleton() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_RadioButton3.default, null));
};

exports.Skeleton = Skeleton;
Skeleton.storyName = 'skeleton';
Skeleton.parameters = {
  info: {
    text: "\n        Placeholder skeleton state to use when content is loading.\n      "
  }
};
