// carbon-components-react/lib/components/Loading/Loading-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonKnobs = require("@storybook/addon-knobs");

var _Loading = _interopRequireDefault(require("../Loading"));

var _Loading2 = _interopRequireDefault(require("./Loading.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var props = function props() {
  return {
    active: (0, _addonKnobs.boolean)('Active (active)', true),
    withOverlay: (0, _addonKnobs.boolean)('With overlay (withOverlay)', false),
    small: (0, _addonKnobs.boolean)('Small (small)', false),
    description: (0, _addonKnobs.text)('Description (description)', 'Active loading indicator')
  };
};

var _default = {
  title: 'Loading',
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    component: _Loading.default,
    docs: {
      page: _Loading2.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Loading.default, _extends({}, props(), {
    className: 'some-class'
  }));
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n        Loading spinners are used when retrieving data or performing slow computations,\n        and help to notify users that loading is underway. The 'active' property is true by default;\n        set to false to end the animation.\n      "
  }
};
