// carbon-components-react/lib/components/OrderedList/OrderedList-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NativeListStyles = exports.Nested = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _OrderedList = _interopRequireDefault(require("../OrderedList"));

var _ListItem = _interopRequireDefault(require("../ListItem"));

var _OrderedList2 = _interopRequireDefault(require("./OrderedList.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _default = {
  title: 'OrderedList',
  parameters: {
    component: _OrderedList.default,
    docs: {
      page: _OrderedList2.default
    },
    subcomponents: {
      ListItem: _ListItem.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_OrderedList.default, null, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"));
};

exports.Default = Default;
Default.storyName = 'default';
Default.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};

var Nested = function Nested() {
  return /*#__PURE__*/_react.default.createElement(_OrderedList.default, null, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1", /*#__PURE__*/_react.default.createElement(_OrderedList.default, {
    nested: true
  }, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2", /*#__PURE__*/_react.default.createElement(_OrderedList.default, {
    nested: true
  }, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 3"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 3"))))), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"));
};

exports.Nested = Nested;
Nested.storyName = 'nested';
Nested.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};

var NativeListStyles = function NativeListStyles() {
  return /*#__PURE__*/_react.default.createElement(_OrderedList.default, {
    native: true
  }, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1", /*#__PURE__*/_react.default.createElement(_OrderedList.default, {
    nested: true
  }, /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 2"))), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"), /*#__PURE__*/_react.default.createElement(_ListItem.default, null, "Ordered List level 1"));
};

exports.NativeListStyles = NativeListStyles;
Default.storyName = 'native styles';
NativeListStyles.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};
