// carbon-components-react/lib/components/Link/Link-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _Link = _interopRequireDefault(require("../Link"));

var _Link2 = _interopRequireDefault(require("./Link.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

/* eslint-disable no-console */
var props = function props() {
  return {
    className: 'some-class',
    href: (0, _addonKnobs.text)('The link href (href)', '#'),
    inline: (0, _addonKnobs.boolean)('Use the in-line variant (inline)', false),
    visited: (0, _addonKnobs.boolean)('Allow visited styles', false),
    onClick: function (handler) {
      return function (evt) {
        evt.preventDefault(); // Prevent link from being followed for demo purpose

        handler(evt);
      };
    }((0, _addonActions.action)('onClick')),
    disabled: (0, _addonKnobs.boolean)('Disabled', false),
    size: (0, _addonKnobs.select)('Link size', {
      Default: undefined,
      'Small (sm)': 'sm',
      'Large (lg)': 'lg'
    })
  };
};

var _default = {
  title: 'Link',
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    component: _Link.default,
    docs: {
      page: _Link2.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Link.default, props(), "Link");
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n            Links are typically used as a means of navigation either within the application, to a place outside, or to a resource.\n            For anything else, especially things that change data, you should be using a button.\n          "
  }
};
