// carbon-components-react/lib/components/CodeSnippet/CodeSnippet-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.playground = exports.skeleton = exports.singleline = exports.multiline = exports.inline = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _CodeSnippet = _interopRequireDefault(require("../CodeSnippet"));

var _CodeSnippet2 = _interopRequireDefault(require("./CodeSnippet.Skeleton"));

var _CodeSnippet3 = _interopRequireDefault(require("./CodeSnippet.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var _default = {
  title: 'CodeSnippet',
  component: _CodeSnippet.default,
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    docs: {
      page: _CodeSnippet3.default
    }
  }
};
exports.default = _default;

var inline = function inline() {
  return /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    type: "inline",
    feedback: "Copied to clipboard"
  }, 'node -v');
};

exports.inline = inline;

var multiline = function multiline() {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '50%'
    }
  }, /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, _extends({}, props(), {
    type: "multi",
    feedback: "Copied to clipboard"
  }), "  \"scripts\": {\n    \"build\": \"lerna run build --stream --prefix --npm-client yarn\",\n    \"ci-check\": \"carbon-cli ci-check\",\n    \"clean\": \"lerna run clean && lerna clean --yes && rimraf node_modules\",\n    \"doctoc\": \"doctoc --title '## Table of Contents'\",\n    \"format\": \"prettier --write '**/*.{js,md,scss,ts}' '!**/{build,es,lib,storybook,ts,umd}/**'\",\n    \"format:diff\": \"prettier --list-different '**/*.{js,md,scss,ts}' '!**/{build,es,lib,storybook,ts,umd}/**' '!packages/components/**'\",\n    \"lint\": \"eslint actions config codemods packages\",\n    \"lint:styles\": \"stylelint '**/*.{css,scss}' --report-needless-disables --report-invalid-scope-disables\",\n    \"sync\": \"carbon-cli sync\",\n    \"test\": \"cross-env BABEL_ENV=test jest\",\n    \"test:e2e\": \"cross-env BABEL_ENV=test jest --testPathPattern=e2e --testPathIgnorePatterns='examples,/packages/components/,/packages/react/'\"\n  },\n  \"resolutions\": {\n    \"react\": \"~16.9.0\",\n    \"react-dom\": \"~16.9.0\",\n    \"react-is\": \"~16.9.0\",\n    \"react-test-renderer\": \"~16.9.0\"\n  },\n  \"devDependencies\": {\n    \"@babel/core\": \"^7.10.0\",\n    \"@babel/plugin-proposal-class-properties\": \"^7.7.4\",\n    \"@babel/plugin-proposal-export-default-from\": \"^7.7.4\",\n    \"@babel/plugin-proposal-export-namespace-from\": \"^7.7.4\",\n    \"@babel/plugin-transform-runtime\": \"^7.10.0\",\n    \"@babel/preset-env\": \"^7.10.0\",\n    \"@babel/preset-react\": \"^7.10.0\",\n    \"@babel/runtime\": \"^7.10.0\",\n    \"@commitlint/cli\": \"^8.3.5\","));
};

exports.multiline = multiline;

var singleline = function singleline() {
  return /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
    type: "single",
    feedback: "Copied to clipboard"
  }, "yarn add carbon-components@latest carbon-components-react@latest @carbon/icons-react@latest carbon-icons@latest");
};

exports.singleline = singleline;

var skeleton = function skeleton() {
  return /*#__PURE__*/_react.default.createElement("div", null, /*#__PURE__*/_react.default.createElement(_CodeSnippet2.default, {
    type: "single",
    style: {
      marginBottom: 8
    }
  }), /*#__PURE__*/_react.default.createElement(_CodeSnippet2.default, {
    type: "multi"
  }));
};

exports.skeleton = skeleton;

var lightPropMessage = /*#__PURE__*/_react.default.createElement("small", {
  style: {
    display: 'block',
    paddingBottom: '1rem'
  }
}, "The snippet container should never be the same color as the page background.", /*#__PURE__*/_react.default.createElement("br", null), 'Do not use the ', /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
  type: "inline",
  light: true
}, "light"), ' variant on ', /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
  type: "inline",
  light: true
}, "$ui-background"), ' or ', /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, {
  type: "inline",
  light: true
}, "$ui-02"), ".");

var props = function props() {
  return {
    type: (0, _addonKnobs.select)('Type', {
      inline: 'inline',
      'single line': 'single',
      'multiple line': 'multi'
    }, 'inline'),
    light: (0, _addonKnobs.boolean)('Light variant', false),
    feedback: (0, _addonKnobs.text)('Feedback text', 'Copied to clipboard'),
    showMoreText: (0, _addonKnobs.text)('Text for "show more" button', 'Show more'),
    showLessText: (0, _addonKnobs.text)('Text for "show less" button', 'Show less'),
    hideCopyButton: (0, _addonKnobs.boolean)('Hide copy button', false),
    onClick: (0, _addonActions.action)('onClick'),
    copyButtonDescription: (0, _addonKnobs.text)('Copy button title', 'Copy code snippet'),
    ariaLabel: (0, _addonKnobs.text)('ARIA label', 'Container label'),
    wrapText: (0, _addonKnobs.boolean)('Wrap text', true)
  };
};

var playground = function playground() {
  return /*#__PURE__*/_react.default.createElement("div", {
    className: props().light ? 'bx--tile' : ''
  }, props().light && lightPropMessage, /*#__PURE__*/_react.default.createElement(_CodeSnippet.default, props(), props().type === 'multi' ? "@mixin grid-container {\n  width: 100%;\n  padding-right: padding(mobile);\n  padding-left: padding(mobile);\n}" : 'node -v'));
};

exports.playground = playground;
