// carbon-components-react/lib/components/Toolbar/Toolbar-story.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _iconsReact = require("@carbon/icons-react");

var _Toolbar = _interopRequireWildcard(require("../Toolbar"));

var _OverflowMenu = _interopRequireDefault(require("../OverflowMenu"));

var _OverflowMenuItem = _interopRequireDefault(require("../OverflowMenuItem"));

var _Checkbox = _interopRequireDefault(require("../Checkbox"));

var _RadioButton = _interopRequireDefault(require("../RadioButton"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var toolbarProps = {
  className: 'some-class'
};
var inputProps = {
  className: 'some-class',
  onChange: (0, _addonActions.action)('onChange')
};
var _default = {
  title: 'Toolbar [Deprecated]',
  parameters: {
    component: _Toolbar.default,
    subcomponents: {
      ToolbarItem: _Toolbar.ToolbarItem,
      ToolbarTitle: _Toolbar.ToolbarTitle,
      ToolbarOption: _Toolbar.ToolbarOption,
      ToolbarDivider: _Toolbar.ToolbarDivider
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Toolbar.default, _extends({}, toolbarProps, {
    className: "some-class"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, {
    type: "search",
    placeHolderText: "Search"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, null, /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, {
    renderIcon: _iconsReact.Filter16
  }, /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarTitle, {
    title: "FILTER BY"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, _extends({}, inputProps, {
    id: "opt-1",
    labelText: "Filter option 1"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, _extends({}, inputProps, {
    id: "opt-2",
    labelText: "Filter option 2"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_Checkbox.default, _extends({}, inputProps, {
    id: "opt-3",
    labelText: "Filter option 3"
  }))))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarItem, null, /*#__PURE__*/_react.default.createElement(_OverflowMenu.default, null, /*#__PURE__*/_react.default.createElement(_OverflowMenuItem.default, {
    itemText: "Refresh table"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarDivider, null), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarTitle, {
    title: "ROW HEIGHT"
  }), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_RadioButton.default, _extends({}, inputProps, {
    value: "short",
    id: "radio-1",
    name: "toolbar-radio",
    labelText: "Short"
  }))), /*#__PURE__*/_react.default.createElement(_Toolbar.ToolbarOption, null, /*#__PURE__*/_react.default.createElement(_RadioButton.default, _extends({}, inputProps, {
    value: "tall",
    id: "radio-2",
    name: "toolbar-radio",
    labelText: "Tall"
  }))))));
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n      Toolbar stuff\n    "
  }
};
