// carbon-components-react/lib/components/Search/Search-story.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Skeleton = exports.DeprecatedCustomButtons = exports.Default = exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _addonKnobs = require("@storybook/addon-knobs");

var _Search = _interopRequireDefault(require("../Search"));

var _Search2 = _interopRequireDefault(require("../Search/Search.Skeleton"));

var _SearchFilterButton = _interopRequireDefault(require("../SearchFilterButton"));

var _SearchLayoutButton = _interopRequireDefault(require("../SearchLayoutButton"));

var _Search3 = _interopRequireDefault(require("./Search.mdx"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var sizes = {
  'Regular size (xl)': 'xl',
  'Large size (lg)': 'lg',
  'Small size (sm)': 'sm'
};

var props = function props() {
  return {
    className: 'some-class',
    size: (0, _addonKnobs.select)('Size (size)', sizes, 'xl'),
    light: (0, _addonKnobs.boolean)('Light variant (light)', false),
    disabled: (0, _addonKnobs.boolean)('Disabled (disabled)', false),
    name: (0, _addonKnobs.text)('Form item name (name)', ''),
    defaultValue: (0, _addonKnobs.text)('Default value (defaultValue)', ''),
    labelText: (0, _addonKnobs.text)('Label text (labelText)', 'Search'),
    closeButtonLabelText: (0, _addonKnobs.text)('The label text for the close button (closeButtonLabelText)', 'Clear search input'),
    placeHolderText: (0, _addonKnobs.text)('Placeholder text (placeHolderText)', 'Search'),
    onChange: (0, _addonActions.action)('onChange'),
    onKeyDown: (0, _addonActions.action)('onKeyDown')
  };
};

var _default = {
  title: 'Search',
  decorators: [_addonKnobs.withKnobs],
  parameters: {
    component: _Search.default,
    docs: {
      page: _Search3.default
    },
    subcomponents: {
      SearchSkeleton: _Search2.default,
      SearchFilterButton: _SearchFilterButton.default,
      SearchLayoutButton: _SearchLayoutButton.default
    }
  }
};
exports.default = _default;

var Default = function Default() {
  return /*#__PURE__*/_react.default.createElement(_Search.default, _extends({}, props(), {
    id: "search-1"
  }));
};

exports.Default = Default;
Default.parameters = {
  info: {
    text: "\n            Search enables users to specify a word or a phrase to find particular relevant pieces of content\n            without the use of navigation. Search can be used as the primary means of discovering content,\n            or as a filter to aid the user in finding content.\n          "
  }
};

var DeprecatedCustomButtons = function DeprecatedCustomButtons() {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      display: 'flex'
    }
  }, /*#__PURE__*/_react.default.createElement(_Search.default, _extends({}, props(), {
    id: "search-1"
  })), /*#__PURE__*/_react.default.createElement(_SearchFilterButton.default, {
    onClick: (0, _addonActions.action)('onClick')
  }), /*#__PURE__*/_react.default.createElement(_SearchLayoutButton.default, {
    onClick: (0, _addonActions.action)('onClick')
  }));
};

exports.DeprecatedCustomButtons = DeprecatedCustomButtons;
DeprecatedCustomButtons.storyName = '[Deprecated] custom buttons';
DeprecatedCustomButtons.parameters = {
  info: {
    text: "\n        You can control what set of buttons you want.\n      "
  }
};

var Skeleton = function Skeleton() {
  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      width: '200px'
    }
  }, /*#__PURE__*/_react.default.createElement(_Search2.default, null), "\xA0", /*#__PURE__*/_react.default.createElement(_Search2.default, {
    small: true
  }));
};

exports.Skeleton = Skeleton;
Skeleton.storyName = 'skeleton';
Skeleton.parameters = {
  info: {
    text: "\n        Placeholder skeleton state to use when content is loading.\n      "
  }
};
