// carbon-components-react/es/components/ComposedModal/ComposedModal-story.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { if (typeof Symbol === "undefined" || !(Symbol.iterator in Object(arr))) return; var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, select, text } from '@storybook/addon-knobs';
import { settings } from 'carbon-components';
import ComposedModal, { ModalHeader, ModalBody, ModalFooter } from '../ComposedModal';
import Button from '../Button';
import mdx from './ComposedModal.mdx';
var prefix = settings.prefix;
var sizes = {
  Default: '',
  'Extra small (xs)': 'xs',
  'Small (sm)': 'sm',
  'Large (lg)': 'lg'
};
var props = {
  composedModal: function composedModal() {
    var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
        titleOnly = _ref.titleOnly;

    return {
      open: boolean('Open (open in <ComposedModal>)', true),
      onKeyDown: action('onKeyDown'),
      danger: boolean('Danger mode (danger)', false),
      selectorPrimaryFocus: text('Primary focus element selector (selectorPrimaryFocus)', '[data-modal-primary-focus]'),
      size: select('Size (size)', sizes, titleOnly ? 'sm' : ''),
      preventCloseOnClickOutside: boolean('Prevent closing on click outside of modal (preventCloseOnClickOutside)', false)
    };
  },
  composedModalWithLauncher: function composedModalWithLauncher() {
    var _ref2 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
        titleOnly = _ref2.titleOnly;

    return {
      onKeyDown: action('onKeyDown'),
      danger: boolean('Danger mode (danger)', false),
      selectorPrimaryFocus: text('Primary focus element selector (selectorPrimaryFocus)', '[data-modal-primary-focus]'),
      size: select('Size (size)', sizes, titleOnly ? 'sm' : ''),
      preventCloseOnClickOutside: boolean('Prevent closing on click outside of modal (preventCloseOnClickOutside)', false)
    };
  },
  modalHeader: function modalHeader() {
    var _ref3 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
        titleOnly = _ref3.titleOnly;

    return {
      label: text('Optional Label (label in <ModalHeader>)', 'Optional Label'),
      title: text('Optional title (title in <ModalHeader>)', titleOnly ? "\n      Passive modal title as the message. Should be direct and 3 lines or less.\n    ".trim() : 'Example'),
      iconDescription: text('Close icon description (iconDescription in <ModalHeader>)', 'Close'),
      buttonOnClick: action('buttonOnClick'),
      preventCloseOnClickOutside: boolean('Prevent closing on click outside of modal (preventCloseOnClickOutside)', false)
    };
  },
  modalBody: function modalBody() {
    return {
      hasScrollingContent: boolean('Modal contains scrollable content (hasScrollingContent)', true),
      'aria-label': text('ARIA label for content', 'Example modal content')
    };
  },
  modalFooter: function modalFooter() {
    return {
      primaryButtonText: text('Primary button text (primaryButtonText in <ModalFooter>)', 'Save'),
      primaryButtonDisabled: boolean('Primary button disabled (primaryButtonDisabled in <ModalFooter>)', false),
      secondaryButtonText: text('Secondary button text (secondaryButtonText in <ModalFooter>)', ''),
      onRequestClose: action('onRequestClose'),
      onRequestSubmit: action('onRequestSubmit')
    };
  }
};
var scrollingContent = /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("h3", null, "Lorem ipsum"), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '), /*#__PURE__*/React.createElement("p", null, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum.", ' '));
/**
 * Simple state manager for modals.
 */

var ModalStateManager = function ModalStateManager(_ref4) {
  var LauncherContent = _ref4.renderLauncher,
      ModalContent = _ref4.children;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      open = _useState2[0],
      setOpen = _useState2[1];

  return /*#__PURE__*/React.createElement(React.Fragment, null, !ModalContent || typeof document === 'undefined' ? null : /*#__PURE__*/ReactDOM.createPortal( /*#__PURE__*/React.createElement(ModalContent, {
    open: open,
    setOpen: setOpen
  }), document.body), LauncherContent && /*#__PURE__*/React.createElement(LauncherContent, {
    open: open,
    setOpen: setOpen
  }));
};

export default {
  title: 'ComposedModal',
  decorators: [withKnobs],
  parameters: {
    component: ComposedModal,
    subcomponents: {
      ModalHeader: ModalHeader,
      ModalBody: ModalBody,
      ModalFooter: ModalFooter
    },
    docs: {
      page: mdx
    }
  }
};
export var UsingHeaderFooterProps = function UsingHeaderFooterProps() {
  var _props$composedModalW = props.composedModalWithLauncher(),
      size = _props$composedModalW.size,
      rest = _objectWithoutProperties(_props$composedModalW, ["size"]);

  var _props$modalBody = props.modalBody(),
      hasScrollingContent = _props$modalBody.hasScrollingContent,
      bodyProps = _objectWithoutProperties(_props$modalBody, ["hasScrollingContent"]);

  return /*#__PURE__*/React.createElement(ComposedModal, _extends({}, rest, {
    danger: true,
    size: size || undefined
  }), /*#__PURE__*/React.createElement(ModalHeader, props.modalHeader()), /*#__PURE__*/React.createElement(ModalBody, _extends({}, bodyProps, {
    "aria-label": hasScrollingContent ? 'Modal content' : undefined
  }), /*#__PURE__*/React.createElement("p", {
    className: "".concat(prefix, "--modal-content__text")
  }, "Please see the \"With Trigger Button\" storybook example for a demo of this functionality."), /*#__PURE__*/React.createElement("br", null), hasScrollingContent && scrollingContent), /*#__PURE__*/React.createElement(ModalFooter, props.modalFooter()));
};
UsingHeaderFooterProps.storyName = 'Using Header / Footer Props';
UsingHeaderFooterProps.parameters = {
  info: {
    text: "\n        Composed Modal allows you to create your own modal with just the parts you need. The ComposedModal element provides the state management for open/close, as well as passes the ModalHeader a prop to close the modal (with the close button).\n\n        The interior components - ModalHeader / ModalBody / ModalFooter - are all container elements that will render any children you add in, wrapped in the appropriate CSS classes.\n\n        The Modal Header / Modal Footer come with some built in props to let you accelerate towards standard Carbon modal UI. If there are customizations you need to do, see the next example of just using the interior components as containers.\n      "
  }
};
export var UsingChildNodes = function UsingChildNodes() {
  var _props$composedModal = props.composedModal(),
      size = _props$composedModal.size,
      rest = _objectWithoutProperties(_props$composedModal, ["size"]);

  var _props$modalBody2 = props.modalBody(),
      hasScrollingContent = _props$modalBody2.hasScrollingContent,
      bodyProps = _objectWithoutProperties(_props$modalBody2, ["hasScrollingContent"]);

  return /*#__PURE__*/React.createElement(ComposedModal, _extends({}, rest, {
    size: size || undefined
  }), /*#__PURE__*/React.createElement(ModalHeader, props.modalHeader(), /*#__PURE__*/React.createElement("h1", null, "Testing")), /*#__PURE__*/React.createElement(ModalBody, _extends({}, bodyProps, {
    "aria-label": hasScrollingContent ? 'Modal content' : undefined
  }), /*#__PURE__*/React.createElement("p", null, "Please see the \"With Trigger Button\" storybook example for a demo of this functionality."), /*#__PURE__*/React.createElement("br", null), hasScrollingContent && scrollingContent), /*#__PURE__*/React.createElement(ModalFooter, null, /*#__PURE__*/React.createElement(Button, {
    kind: "secondary"
  }, "Cancel"), /*#__PURE__*/React.createElement(Button, {
    kind: props.composedModal().danger ? 'danger' : 'primary'
  }, "Save")));
};
UsingChildNodes.storyName = 'Using child nodes';
UsingChildNodes.parameters = {
  info: {
    text: "\n        Alternatively, you can just use the Modal components as wrapper elements and figure the children out yourself. We do suggest for the header you utilize the built in props for label and title though, for the footer it's mostly a composed element so creating the two buttons yourself (using the Button component) is probably the most straight-forward pattern.\n      "
  }
};
export var TitleOnly = function TitleOnly() {
  var _props$composedModal2 = props.composedModal({
    titleOnly: true
  }),
      size = _props$composedModal2.size,
      rest = _objectWithoutProperties(_props$composedModal2, ["size"]);

  return /*#__PURE__*/React.createElement(ComposedModal, _extends({}, rest, {
    size: size || undefined
  }), /*#__PURE__*/React.createElement(ModalHeader, props.modalHeader({
    titleOnly: true
  })), /*#__PURE__*/React.createElement(ModalBody, null), /*#__PURE__*/React.createElement(ModalFooter, props.modalFooter()));
};
TitleOnly.storyName = 'Title only';
TitleOnly.parameters = {
  info: {
    text: "\n      In \"small\" and \"xs\" modals size, the title is allowed to span multiple lines and be used for the main message.\n      It should be less than 3 lines of text. If more room is required then use the standard body copy format.\n    "
  }
};
export var WithTriggerButton = function WithTriggerButton() {
  var _props$composedModalW2 = props.composedModalWithLauncher(),
      size = _props$composedModalW2.size,
      rest = _objectWithoutProperties(_props$composedModalW2, ["size"]);

  var _props$modalBody3 = props.modalBody(),
      hasScrollingContent = _props$modalBody3.hasScrollingContent,
      bodyProps = _objectWithoutProperties(_props$modalBody3, ["hasScrollingContent"]);

  return /*#__PURE__*/React.createElement(ModalStateManager, {
    renderLauncher: function renderLauncher(_ref5) {
      var setOpen = _ref5.setOpen;
      return /*#__PURE__*/React.createElement(Button, {
        onClick: function onClick() {
          return setOpen(true);
        }
      }, "Launch composed modal");
    }
  }, function (_ref6) {
    var open = _ref6.open,
        setOpen = _ref6.setOpen;
    return /*#__PURE__*/React.createElement(ComposedModal, _extends({}, rest, {
      open: open,
      size: size || undefined,
      onClose: function onClose() {
        return setOpen(false);
      }
    }), /*#__PURE__*/React.createElement(ModalHeader, props.modalHeader()), /*#__PURE__*/React.createElement(ModalBody, _extends({}, bodyProps, {
      "aria-label": hasScrollingContent ? 'Modal content' : undefined
    }), hasScrollingContent && scrollingContent), /*#__PURE__*/React.createElement(ModalFooter, props.modalFooter()));
  });
};
WithTriggerButton.storyName = 'Example usage with trigger button';
WithTriggerButton.parameters = {
  info: {
    text: "\n        An example ComposedModal with a trigger button\n      "
  }
};
