// carbon-components-react/es/components/OrderedList/OrderedList-story.js
import React from 'react';
import OrderedList from '../OrderedList';
import ListItem from '../ListItem';
import mdx from './OrderedList.mdx';
export default {
  title: 'OrderedList',
  parameters: {
    component: OrderedList,
    docs: {
      page: mdx
    },
    subcomponents: {
      ListItem: ListItem
    }
  }
};
export var Default = function Default() {
  return /*#__PURE__*/React.createElement(OrderedList, null, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"));
};
Default.storyName = 'default';
Default.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};
export var Nested = function Nested() {
  return /*#__PURE__*/React.createElement(OrderedList, null, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1", /*#__PURE__*/React.createElement(OrderedList, {
    nested: true
  }, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2", /*#__PURE__*/React.createElement(OrderedList, {
    nested: true
  }, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 3"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 3"))))), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"));
};
Nested.storyName = 'nested';
Nested.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};
export var NativeListStyles = function NativeListStyles() {
  return /*#__PURE__*/React.createElement(OrderedList, {
    native: true
  }, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1", /*#__PURE__*/React.createElement(OrderedList, {
    nested: true
  }, /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 2"))), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"), /*#__PURE__*/React.createElement(ListItem, null, "Ordered List level 1"));
};
Default.storyName = 'native styles';
NativeListStyles.parameters = {
  info: {
    text: "Lists consist of related content grouped together and organized vertically. Ordered lists are used to present content in a numbered list."
  }
};
